03.bowling/bowling.rb

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'rubygems'
+
+obtained_score = ARGV[0]
+scores = obtained_score.split(',')
+shots = []
+
+scores.each do |score|
+  if score == 'X'
+    shots << 10
+    shots << 0
+  else
+    shots << score.to_i
+  end
+end
+
+frames = shots.each_slice(2).to_a
+
+point = 0
+frames.each_with_index do |current_frame, i|
+  next_frame = frames[i + 1]
+  after_next_frame = frames[i + 2]
+  strike = current_frame[0] == 10
+  spare = current_frame.sum == 10
```

Comment:
```suggestion
  spare = !strike && current_frame.sum == 10
```

ボウリングのルールを厳密に表現するなら、スペアとストライクは明確に区別した方がベターかもしれません